Bottom: 3af74542971ae63ba8cb3ceccf9c3c28bd3efaec
Top:    5c417eb90a817c248f8eb45036fed31eb81f5d45
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-18 21:27:59 +0100

Refresh of initialise-the-hid-vtable-by-f

---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 4a6425a..d18cd2a 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -1169,40 +1169,39 @@ gerber_set_crosshair (int x, int y, int action)
 {
 }
 
-
 void
 hid_gerber_init ()
 {
-  memset (&gcode_hid, 0, sizeof (HID));
-
-  gcode_hid.struct_size         = sizeof (HID);
-  gcode_hid.name                = "gerber";
-  gcode_hid.description         = "RS-274X (Gerber) export.";
-  gcode_hid.exporter            = 1;
-  gcode_hid.poly_dicer          = 1;
-
-  gcode_hid.get_export_options  = gerber_get_export_options;
-  gcode_hid.do_export           = gerber_do_export;
-  gcode_hid.parse_arguments     = gerber_parse_arguments;
-  gcode_hid.set_layer           = gerber_set_layer;
-  gcode_hid.make_gc             = gerber_make_gc;
-  gcode_hid.destroy_gc          = gerber_destroy_gc;
-  gcode_hid.use_mask            = gerber_use_mask;
-  gcode_hid.set_color           = gerber_set_color;
-  gcode_hid.set_line_cap        = gerber_set_line_cap;
-  gcode_hid.set_line_width      = gerber_set_line_width;
-  gcode_hid.set_draw_xor        = gerber_set_draw_xor;
-  gcode_hid.set_draw_faded      = gerber_set_draw_faded;
-  gcode_hid.set_line_cap_angle  = gerber_set_line_cap_angle;
-  gcode_hid.draw_line           = gerber_draw_line;
-  gcode_hid.draw_arc            = gerber_draw_arc;
-  gcode_hid.draw_rect           = gerber_draw_rect;
-  gcode_hid.fill_circle         = gerber_fill_circle;
-  gcode_hid.fill_polygon        = gerber_fill_polygon;
-  gcode_hid.fill_pcb_polygon    = common_fill_pcb_polygon;
-  gcode_hid.fill_rect           = gerber_fill_rect;
-  gcode_hid.calibrate           = gerber_calibrate;
-  gcode_hid.set_crosshair       = gerber_set_crosshair;
+  memset (&gerber_hid, 0, sizeof (HID));
+
+  gerber_hid.struct_size         = sizeof (HID);
+  gerber_hid.name                = "gerber";
+  gerber_hid.description         = "RS-274X (Gerber) export.";
+  gerber_hid.exporter            = 1;
+  gerber_hid.poly_dicer          = 1;
+
+  gerber_hid.get_export_options  = gerber_get_export_options;
+  gerber_hid.do_export           = gerber_do_export;
+  gerber_hid.parse_arguments     = gerber_parse_arguments;
+  gerber_hid.set_layer           = gerber_set_layer;
+  gerber_hid.make_gc             = gerber_make_gc;
+  gerber_hid.destroy_gc          = gerber_destroy_gc;
+  gerber_hid.use_mask            = gerber_use_mask;
+  gerber_hid.set_color           = gerber_set_color;
+  gerber_hid.set_line_cap        = gerber_set_line_cap;
+  gerber_hid.set_line_width      = gerber_set_line_width;
+  gerber_hid.set_draw_xor        = gerber_set_draw_xor;
+  gerber_hid.set_draw_faded      = gerber_set_draw_faded;
+  gerber_hid.set_line_cap_angle  = gerber_set_line_cap_angle;
+  gerber_hid.draw_line           = gerber_draw_line;
+  gerber_hid.draw_arc            = gerber_draw_arc;
+  gerber_hid.draw_rect           = gerber_draw_rect;
+  gerber_hid.fill_circle         = gerber_fill_circle;
+  gerber_hid.fill_polygon        = gerber_fill_polygon;
+  gerber_hid.fill_pcb_polygon    = common_fill_pcb_polygon;
+  gerber_hid.fill_rect           = gerber_fill_rect;
+  gerber_hid.calibrate           = gerber_calibrate;
+  gerber_hid.set_crosshair       = gerber_set_crosshair;
 
   apply_default_hid (&gerber_hid, 0);
   hid_register_hid (&gerber_hid);
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index a291f0f..95cb675 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2155,7 +2155,7 @@ hid_gtk_init ()
   ghid_hid.beep                 = ghid_beep;
   ghid_hid.progress             = ghid_progress;
   ghid_hid.drc_gui              = &ghid_drc_gui,
-  ghid_hid.attributes           = ghid_attributes;
+  ghid_hid.attribute_dialog     = ghid_attributes;
 
   hid_register_hid (&ghid_hid);
 #include "gtk_lists.h"
diff --git a/src/hid/lpr/lpr.c b/src/hid/lpr/lpr.c
index 5592f25..f65c5a5 100644
--- a/src/hid/lpr/lpr.c
+++ b/src/hid/lpr/lpr.c
@@ -116,7 +116,7 @@ static HID lpr_hid;
 void
 hid_lpr_init ()
 {
-  memset (lpr_hid, 0, sizeof (HID));
+  memset (&lpr_hid, 0, sizeof (HID));
 
   lpr_hid.struct_size         = sizeof (HID);
   lpr_hid.name                = "lpr";
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index 4376459..e1effd3 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -72,7 +72,6 @@
 #include "hid.h"
 #include "../hidint.h"
 #include "hid/common/draw_helpers.h"
-#include "nelma.h"
 
 #include <gd.h>
 
@@ -107,6 +106,8 @@ struct hid_gc_struct {
 	gdImagePtr      brush;
 };
 
+static HID nelma_hid;
+
 static struct color_struct *black = NULL, *white = NULL;
 static int      linewidth = -1;
 static int      lastgroup = -1;
@@ -1046,8 +1047,6 @@ nelma_set_crosshair(int x, int y, int a)
 
 #include "dolists.h"
 
-static HID nelma_hid;
-
 void
 hid_nelma_init()
 {
diff --git a/src/hid/nelma/nelma.h b/src/hid/nelma/nelma.h
deleted file mode 100644
index 0cbe0d7..0000000
--- a/src/hid/nelma/nelma.h
+++ /dev/null
@@ -1,3 +0,0 @@
-/* $Id$ */
-
-extern HID      nelma_hid;
